### PR TITLE
MC attitude controller: fixes for yaw control

### PR DESCRIPTION
**Pull request by bkueng**
This brings some fixes for the MC attitude controller - in particular it fixes the slow yaw convergence from #7949 (there is one more remaining problem and I'll update the issue).

This does:
- fix the yaw feedforwarding (there was a transformation missing from world to body frame)
- fix computation of yaw weight for attitude control. Previously, the yaw weight was based on the tilt angle of the attitude setpoint (R_sp(2, 2) == cos(tilt angle)). This makes no sense, it means the weight is low for high tilt angles even if there is no roll-pitch error at all.
    This PR changes the weight computation to be based on the tilt angle error: the yaw weight is 1 if there is no roll-pitch error (independent from current tilt angle), and is reduced for higher tilt angle errors.
    The weight was added in 05e9a30573f50dd271f10, without any explanation or derivation of how and why the weight is chosen that way.
- remove the hardcoded max throttle limit of 0.9 for ACRO

Further infos are in the commit messages.

Some test flights:
- https://logs.px4.io/plot_app?log=1adf352b-9e6d-4754-8665-5967c87cf00f
- https://logs.px4.io/plot_app?log=58f771d6-0793-482a-afbe-c12ecfdc748d

These tests include the mixer saturation improvements (https://github.com/PX4/Firmware/pull/7920) and I disabled the roll-pitch correction in presence of large yaw errors (see #7949).
I used a maximum tilt angle of 65 degrees and it flies really well. The only thing is that the yaw can go off if you do very fast roll-pitch changes, but this is expected, since the roll & pitch correction take precedence in these cases.

@MaEtUgR plans to implement his attitude controller, then we can compare both.

I'm aware that these are changes at critical places, so careful reviewing and testing is mandatory.

### CODE REVIEW

File: src/modules/mc_att_control/mc_att_control_params.c [hunk 1]
```diff
@@ -335,11 +339,9 @@ PARAM_DEFINE_FLOAT(MC_PITCHRATE_MAX, 220.0f);
 /**
  * Max yaw rate
  *
- * A value of significantly over 120 degrees per second can already lead to mixer saturation.
- *
  * @unit deg/s
  * @min 0.0
- * @max 360.0
+ * @max 3600.0
```
Comment (LorenzMeier) [inline src/modules/mc_att_control/mc_att_control_params.c]: Should be 1800, as 2000 is the hardware limit. We should bound it to what we can still sense.

File: src/modules/mc_att_control/mc_att_control_params.c [hunk 2]
```diff
@@ -321,11 +323,13 @@ PARAM_DEFINE_FLOAT(MC_ROLLRATE_MAX, 220.0f);
...
  *
- * Limit for pitch rate, has effect for large rotations in autonomous mode, to avoid large control output and mixer saturation.
+ * Limit for pitch rate in manual and auto modes (except acro).
+ * Has effect for large rotations in autonomous mode, to avoid large control
+ * output and mixer saturation.
  *
  * @unit deg/s
  * @min 0.0
- * @max 360.0
+ * @max 3600.0
```
Comment (LorenzMeier) [inline src/modules/mc_att_control/mc_att_control_params.c]: Should be 1800, as 2000 is the hardware limit. We should bound it to what we can still sense.

File: src/modules/mc_att_control/mc_att_control_params.c [hunk 3]
```diff
@@ -307,11 +307,13 @@ PARAM_DEFINE_FLOAT(MC_YAW_FF, 0.5f);
...
  *
- * Limit for roll rate, has effect for large rotations in autonomous mode, to avoid large control output and mixer saturation.
+ * Limit for roll rate in manual and auto modes (except acro).
+ * Has effect for large rotations in autonomous mode, to avoid large control
+ * output and mixer saturation.
  *
  * @unit deg/s
  * @min 0.0
- * @max 360.0
+ * @max 3600.0
```
Comment (LorenzMeier) [inline src/modules/mc_att_control/mc_att_control_params.c]: Should be 1800, as 2000 is the hardware limit. We should bound it to what we can still sense.

Comment (bkueng) [reply src/modules/mc_att_control/mc_att_control_params.c]: Makes sense, I changed it.

**Pull request merged**